Assets/LeapMotion/Core/Scripts/Utils/Utils.cs

```diff
@@ -452,6 +450,75 @@ public static class Utils {
       return v.x + v.y + v.z + v.w;
     }
 
+    /// <summary>
+    /// Returns the largest component of the input vector.
+    /// </summary>
+    public static float CompMax(this Vector2 v) {
+      return Mathf.Max(v.x, v.y);
+    }
+
+    /// <summary>
+    /// Returns the largest component of the input vector.
+    /// </summary>
+    public static float CompMax(this Vector3 v) {
+      return Mathf.Max(Mathf.Max(v.x, v.y), v.z);
+    }
+
+    /// <summary>
+    /// Returns the largest component of the input vector.
+    /// </summary>
+    public static float CompMax(this Vector4 v) {
+      return Mathf.Max(Mathf.Max(Mathf.Max(v.x, v.y), v.z), v.w);
+    }
+
+    /// <summary>
+    /// Returns the smallest component of the input vector.
+    /// </summary>
+    public static float CompMin(this Vector2 v) {
+      return Mathf.Min(v.x, v.y);
+    }
+
+    /// <summary>
+    /// Returns the smallest component of the input vector.
+    /// </summary>
+    public static float CompMin(this Vector3 v) {
+      return Mathf.Min(Mathf.Min(v.x, v.y), v.z);
+    }
+
+    /// <summary>
+    /// Returns the smallest component of the input vector.
+    /// </summary>
+    public static float CompMin(this Vector4 v) {
+      return Mathf.Min(Mathf.Min(Mathf.Min(v.x, v.y), v.z), v.w);
+    }
+
+    #endregion
+
+    #region Unity Object Utils
+
+    /// <summary>
+    /// Usage is the same as FindObjectOfType, but this method will also return objects
+    /// that are inactive.
+    /// 
+    /// Use this method to search for singleton-pattern objects even if they are disabled,
+    /// but be warned that it's not cheap to call!
+    /// </summary>
+    public static T FindObjectInHierarchy<T>() where T : UnityEngine.Object {
+      T obj = Resources.FindObjectsOfTypeAll<T>().Query().FirstOrDefault();
+      if (obj == null) return null;
+
+#if UNITY_EDITOR
+      // Exclude prefabs.
+      var prefabType = UnityEditor.PrefabUtility.GetPrefabType(obj);
```

Comment:
So this method will return null, even if the requested object _does exist_, if the Resources.Find method has the misfortune of grabbing a prefab.